### PR TITLE
Feat/update testnet subql

### PR DESCRIPTION
**Pull request by mzxyz**

### CODE REVIEW

File: project-testnet.yaml [hunk 1]
```diff
@@ -157,45 +159,40 @@ dataSources:
...
       file: ./dist/index.js
       handlers:
-        - handler: handleNewQuery
+        - handler: handleNewProject
           kind: ethereum/LogHandler
           filter:
             topics:
-              - CreateQuery(uint256 queryId, address creator, bytes32 metadata, bytes32 deploymentId, bytes32 version)
-        - handler: handleUpdateQueryMetadata
+              - event CreateProject(address indexed creator, uint256 indexed projectId, string projectMetadata, ProjectType projectType, bytes32 deploymentId, bytes32 deploymentMetadata)
```
Comment (ianhe8x) [inline project-testnet.yaml]: enum in event can not be correctly encoded, use `uint8` instead
```suggestion
              - event CreateProject(address indexed creator, uint256 indexed projectId, string projectMetadata, uint8 projectType, bytes32 deploymentId, bytes32 deploymentMetadata)
```

File: project-testnet.yaml [hunk 2]
```diff
@@ -157,45 +159,40 @@ dataSources:
...
           kind: ethereum/LogHandler
           filter:
             topics:
-              - UpdateIndexingStatusToReady(address indexer, bytes32 deploymentId)
-        - handler: handleStopIndexing
-          kind: ethereum/LogHandler
-          filter:
-            topics:
-              - StopIndexing(address indexer, bytes32 deploymentId)
+              - ServiceStatusChanged(address indexed indexer, bytes32 indexed deploymentId, ServiceStatus status)
```
Comment (ianhe8x) [inline project-testnet.yaml]:
```suggestion
              - ServiceStatusChanged(address indexed indexer, bytes32 indexed deploymentId, uint8 status)
```

File: project-testnet.yaml [hunk 3]
```diff
@@ -406,13 +408,9 @@ dataSources:
             topics:
               - QuotaAdded(address token, address account, uint256 amount)
   - kind: ethereum/Runtime
-    startBlock: 39105463
```
Comment (ianhe8x) [inline project-testnet.yaml:409]: if you remove the custom startBlock, add anchor back. `<<: *ethereum`

File: src/mappings/projectRegistry.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    deploymentId,
+    deploymentMetadata,
+  } = event.args;
+  const type = projectType as unknown as ProjectType;
+
+  const project = Project.create({
+    id: projectId.toHexString(),
+    owner: creator,
+    type,
+    metadata: projectMetadata,
```
Comment (ianhe8x) [inline src/mappings/projectRegistry.ts:46]: remove `ipfs://` prefix, or other places need to adapt to the difference.

**Review by ianhe8x — CHANGES_REQUESTED** [review]

File: src/mappings/projectRegistry.ts (hunk 4 above)
Comment (mzxyz) [reply src/mappings/projectRegistry.ts:46]: `projectMetadata` doesn't include `ipfs://`, only user call `tokenURI` will contain the base url

**Review by ianhe8x — APPROVED** [review]

**Pull request merged**